### ISSUE TITLE
Embed Donation form - Stripe API integration

### ISSUE THREAD

**Issue by seraph1201** [codersforcauses/starick#126]
## Basic Information
**[Give a brief summary of what the issue is about]**
- Create a form that offers donors the option to choose between making a one-off payment or setting up a monthly recurring donation.
- Integrate with Stripe's API

## Other Information
**[Are there any additional information that we should know about?]**
- Sample donation form: https://luma.org.au/get-involved/donate/
- Figma: https://www.figma.com/file/eNA06epSMBC9n79F4aF3Zs/Starick?type=design&node-id=0-286&mode=design

**Comment by makon1234** [comment]
Hi Aifert, thaanks for fixing up the frontend. The checkout session isn't working anymore, it was before, do you have any idea what the problem is?

**Comment by Aifert** [comment]
> Hi Aifert, thaanks for fixing up the frontend. The checkout session isn't working anymore, it was before, do you have any idea what the problem is?

Hey, previously, in the donation-stripe.tsx, I saw the amount to be checked out and the interval was being hard coded in, so I've since made a amount useState hook to register the amount selected, and also the interval. These values are working fine in the api when logged. I didn't change much of what was previously written with the api calls, only thing I did was added a body so the api can receive the amount and interval being chosen. Where did you get the apikey before? Right now my logs are telling me the apikey is invalid.
